aten/src/ATen/native/cuda/Normalization.cu

```diff
@@ -423,7 +429,7 @@ std::tuple<Tensor&, Tensor&, Tensor&> batch_norm_cuda_out(const Tensor& self, co
 }
 
 std::tuple<Tensor, Tensor, Tensor> batch_norm_cuda(const Tensor& self, const c10::optional<Tensor>& weight_opt, const c10::optional<Tensor>& bias_opt, const c10::optional<Tensor>& running_mean_opt, const c10::optional<Tensor>& running_var_opt, bool train, double momentum, double epsilon) {
-  auto output = at::empty_like(self, self.suggest_memory_format());
+  auto output = at::empty({0}, self.options());
```

Comment:
I think it's better to just use `empty_like` here, without specifying memory format. `empty_like` propagates permutation, so most regular tensors (including the one from the issue) will have strides same as `self`, and will pass `is_contiguous` check. Then you wouldn't need special logic in `batch_norm_elementwise`. On the other hand, `resize_impl_cuda_(out.unsafeGetTensorImpl(), self.sizes(), self.strides());` that you have now can allocate non-memory-dense or self-overlapping tensor, neither of which is good.

Reply:
Okay, returned to call `empty_like` but we still need `resize_impl_cuda_` for the out-argument overload. Note that it can't create non-dense tensors because the channels last implementation only supports dense tensors as input.